### PR TITLE
Fixed Freeze of UI on TextToPDf

### LINKED ISSUE #719: UI freeze when converting text/doc to PDF
UI freezes when we try to convert big text/PDF files to PDF

### PR DESCRIPTION
**Pull request by jddeep**
# Description
Fixed Freeze of UI on TextToPDf.

Screenshot_20190422-193539
Screenshot_20190422-193548

Fixes #719 

## Type of change
Just put an x in the [] which are valid.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by jddeep** [comment]
@Swati4star @vyankatesh24 Please review.

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 1]
```diff
@@ -442,9 +431,12 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                     showSnackbar(mActivity, R.string.text_file_selected);
                     String fileName = mFileUtils.getFileName(mTextFileUri);
                     if (fileName != null) {
-                        if (fileName.endsWith(Constants.textExtension)) mFileExtension = Constants.textExtension;
-                        else if (fileName.endsWith(Constants.docxExtension)) mFileExtension = Constants.docxExtension;
-                        else if (fileName.endsWith(Constants.docExtension)) mFileExtension = Constants.docExtension;
+                        if (fileName.endsWith(Constants.textExtension))
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: Revert these changes

File: app/src/main/java/swati4star/createpdf/util/TextToPdfAsync.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+public class TextToPdfAsync extends AsyncTask<Object, Object, Object> {
+    private PDFUtils mFileUtil;
+    private TextToPDFOptions mTextToPdfOptions;
+    private String mRealPath;
+    private String mFileExtension;
+    private final OnPDFCreatedInterface mOnPDFCreatedInterface;
+    private boolean mSuccess;
+
+    public TextToPdfAsync(String realpath, PDFUtils fileutil, TextToPDFOptions textToPDFOptions,
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/TextToPdfAsync.java]: comments

**Comment by srv-twry** [comment]
@jddeep Do you have some experience with using RxJava? I would highly recommend exploring that library and learning some reactive programming. I would suggest using RxJava/RxAndroid instead of AsyncTask.

https://medium.com/code-yoga/using-rxjava-instead-asynctask-3052697c0f48

File: app/src/main/java/swati4star/createpdf/util/TextToPdfAsync.java (hunk 2 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/util/TextToPdfAsync.java]: added comments

**Comment by jddeep** [comment]
@Swati4star @vyankatesh24 Changes made. Please review,

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 3]
```diff
@@ -517,4 +510,32 @@ private boolean isPermissionGranted() {
...
+            mMaterialDialog.dismiss();
+        if (!success) {
+            showSnackbar(mActivity, R.string.error_occurred);
+            mMorphButtonUtility.morphToGrey(mCreateTextPdf, mMorphButtonUtility.integer());
+            mCreateTextPdf.setEnabled(false);
+            mTextFileUri = null;
+            mButtonClicked = 0;
+            return;
+        }
+        final String finalMPath = mPath;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: what is the need of `finalMPath`?

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 4]
```diff
@@ -517,4 +510,32 @@ private boolean isPermissionGranted() {
...
     }
+
+    @Override
+    public void onPDFCreationStarted() {
+        mMaterialDialog = createAnimationDialog(mActivity);
+        mMaterialDialog.show();
+    }
+
+    @Override
+    public void onPDFCreated(boolean success, String path) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: what is the need of this path variable? We are not using it anywhere

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java (hunk 3 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: it is used. Please see.

File: app/src/main/java/swati4star/createpdf/util/TextToPdfAsync.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            mSuccess = false;
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    protected void onPostExecute(Object o) {
+        super.onPostExecute(o);
+        mOnPDFCreatedInterface.onPDFCreated(mSuccess, mRealPath);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/TextToPdfAsync.java]: we are supposed to return the path where the PDF is created. you are just returning the path which is being sent out as parameter

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java (hunk 3 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: I mean I understand there is no need of that

File: app/src/main/java/swati4star/createpdf/util/TextToPdfAsync.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    private String mRealPath;
+    private String mFileExtension;
+    private final OnPDFCreatedInterface mOnPDFCreatedInterface;
+    private boolean mSuccess;
+
+    /**
+     * This is a public constructor responsible for initializing the path of the actual
+     * file, the PDFUtils instance for the file, the options for text to Pdf, the file
+     * extension, and the onPDFCreatedInterface instance.
+     * @param realpath is the path of the actual file.
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/TextToPdfAsync.java]: path of which file? -> PDF file or text file?

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java (hunk 3 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: But it was present in that fashion earlier so I didn't change that.

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: then remove it

**Comment by jddeep** [comment]
@Swati4star Changes made!

**Pull request merged**